modules/auxiliary/fileformat/badpdf.rb

```diff
@@ -0,0 +1,164 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::FILEFORMAT
+
+  def initialize(info={})
+    super( update_info( info,
+        'Name'          => 'BADPDF Malicious PDF Creator',
+        'Description'   => %q{
+          This module can either creates a blank PDF file which contains a UNC link which can be used
+          to capture NetNTLM credentials, or if the PDFINJECT option is used it will inject the necessary
+          code into an existing PDF document if possible.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        =>
+            [
+              'Assaf Baharav',    # Code provided as POC by CheckPoint
+              'Yaron Fruchtmann', # Code provided as POC by CheckPoint
+              'Ido Solomon',      # Code provided as POC by CheckPoint
+              'Richard Davy - secureyourit.co.uk',  # Metasploit
+            ],
+        'Platform'      => [ 'win' ],
+        'References'    =>
+        [
+          ['CVE', '2018-4993'],
+          ['URL', 'https://research.checkpoint.com/ntlm-credentials-theft-via-pdf-files/']
+        ]
+
+      ))
+    register_options(
+      [
+        OptAddress.new("LHOST", [ true, "Host listening for incoming SMB/WebDAV traffic", nil]),
+        OptString.new("FILENAME", [ false, "Filename"]),
+        OptPath.new("PDFINJECT", [ false, "Path and filename to existing PDF to inject UNC link code into"]),
+
+      ])
+  end
+
+  def run
+    if !datastore['PDFINJECT'].nil? && datastore['PDFINJECT'].to_s.end_with?('.pdf')
+        injectpdf
+    else
+        if !datastore['FILENAME'].nil? && datastore['FILENAME'].to_s.end_with?('.pdf')
+          createpdf
+        else
+          print_error "FILENAME is empty, please enter FILENAME and rerun module"
```

Comment:
This isn't intuitive, as the user could enter a valid filename that doesn't end with `.pdf`, and would be advised that `FILENAME is empty`, leading to confusion.

Here's some psuedo code which might help. You may or may not want to reconsider the requirement for the filename to end with `.pdf`

```ruby
  def run
    if !datastore['PDFINJECT'].nil? && datastore['PDFINJECT'].to_s.end_with?('.pdf')
      injectpdf
    elsif !datastore['FILENAME'].nil? && datastore['FILENAME'].to_s.end_with?('.pdf')
      createpdf
    else
      print_error "FILENAME must end with '.pdf' file extension"
    end
  end
```

Reply:
Done.